eclass/tree-sitter-grammar.eclass

```diff
@@ -36,6 +36,47 @@ RESTRICT+=" !test? ( test )"
 # Used to override upstream tag name if tagged differently, e.g. most releases
 # are v${PV} but some are tagged as rust-${PV}.
 
+# @ECLASS_VARIABLE: TS_BINDINGS
+# @PRE_INHERIT
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Array of bindings language to build. Currently only "python" is supported.
+
+_tree_sitter_set_globals() {
+	local binding
+	for binding in "${TS_BINDINGS[@]}"; do
+		case ${binding} in
+			python)
+				DISTUTILS_EXT=1
+				DISTUTILS_OPTIONAL=1
+				DISTUTILS_USE_PEP517=setuptools
+				PYTHON_COMPAT=( python3_{10..12} )
+				inherit distutils-r1
```

Comment:
I'd be worried about calling `inherit` from inside a function.

Reply:
What exactly is the concern here? That variables are marked as local within global scope on the inherited eclass, shouldn't be one. It could clash with local variables in the tree-sitter eclass but that's an interface documentation problem, not an intrinsic one to inherit.

Reply:
```
<@mgorny> ulm: https://github.com/gentoo/gentoo/pull/35750/commits/7d8de2e7ecb10aad5152be232146c52074e0f682#diff-9377b7fee5bc9b263af528e66cf0c8ab2ae81c5ff172ab1bb4c9af77384b18c6R54 i wonder if it's safe to call inherit from inside a function
<@ulm> mgorny: the spec says that inherit must be called in global scope, not in function scope
```

I guess I'll need to do it outside the function in global scope, and just not forget to cleanup variables after myself

Reply:
Extracted it from the function, and changed the `unset` to be as clean as possible.